src/shackle_backlog.erl

```diff
@@ -24,7 +24,7 @@ check(ServerName, BacklogSize) ->
     check(ServerName, BacklogSize, ?DEFAULT_INCREMENT).
 
 -spec check(server_name(), backlog_size(), pos_integer()) ->
-    boolean().
+    pos_integer() | false.
```

Comment:
I'd like to keep the same interface for `check/3`. I suggest you create another method (e.g. `increment/3`) that get wrapped by `check/3`.